packages/next/build/webpack-config.ts

```diff
@@ -176,6 +177,116 @@ export default async function getBaseWebpackConfig(
 
   const devtool = dev ? 'cheap-module-source-map' : false
 
+  // Contains various versions of the Webpack SplitChunksPlugin used in different build types
+  const splitChunksConfigs: {
+    [propName: string]: webpack.Options.SplitChunksOptions
+  } = {
+    dev: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+      },
+    },
+    selective: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prod: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+        },
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prodGranular: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        framework: {
+          name: 'framework',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom|scheduler|prop-types)[\\/]/,
+          priority: 40,
+        },
+        lib: {
+          test(module: { size: Function }): boolean {
+            return module.size() > 160000
+          },
+          name(module: { identifier: Function; rawRequest: string }): string {
+            const rawRequest =
+              module.rawRequest && module.rawRequest.replace(/^@(\w+)\//, '$1-')
+            if (rawRequest) return rawRequest
+
+            const identifier = module.identifier()
+            const trimmedIdentifier = /(?:^|\/)node_modules\/(.*)/.exec(
+              identifier
+            )
+            const processedIdentifier =
+              trimmedIdentifier &&
+              trimmedIdentifier[1].replace(/^@(\w+)\//, '$1-')
+
+            return processedIdentifier || identifier
+          },
+          priority: 30,
+          minChunks: 1,
+          reuseExistingChunk: true,
+        },
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages,
+          priority: 20,
+        },
+        shared: {
+          name(module, chunks) {
+            return crypto
+              .createHash('sha1')
+              .update(
+                chunks.reduce((acc: string, chunk: any) => {
+                  return acc + chunk.name
+                }, '')
+              )
+              .digest('base64')
+              .replace(/\//g, '')
+          },
+          priority: 10,
+          minChunks: 2,
+          reuseExistingChunk: true,
+        },
+      },
+      maxInitialRequests: 20,
+    },
+  }
+
+  //Select appropriate SplitChunksPlugin config for this build
```

Comment:
```suggestion
  // Select appropriate SplitChunksPlugin config for this build
```

nit because I can't unsee this